src/components/TextInput/index.js

```diff
@@ -29,11 +31,19 @@ export default function TextInput({
   input.classList.add(variation);
   input.style.backgroundImage = `url(${assetUrl})`;
   input.classList.add(assetPosition);
+  if (value !== '') {
+    this.setValue(value);
+  }
 
   input.addEventListener('focus', () => {
     if (input.disabled) return;
     input.classList.remove('input-error');
   });
+
+  input.addEventListener('blur', () => {
+    if (input.disabled) return;
+    this.emit('value:changed', input.value);
+  });
```

Comment:
Não gosto muito desse `blur`, acredito que seria melhor um evento de `input`

Também não acho que precise desse `if` se o input ta disabled já não é possível selecionar ele, acho que essa função poderia ser:

```suggestion
  input.addEventListener('blur', () => this.emit('value:changed', input.value));
```

Reply:
Eu tentei fazer com o input antes mas aí o input fica com um comportamente estranho que não consegui entender o porque. Ele só deixa digitar um número e depois não consigo nem apagar o input mais. Por exemplo: se eu digitar 4, ele fica 4.0 e eu não consigo digitar e nem apagar mais.